### ISSUE TITLE
Pie Chart: add support for slice labels outside the pie

### ISSUE THREAD

**Issue by cpa-level-it** [ScottPlot/ScottPlot#2275]
## Feature Suggestion
 
Alter the piechart API to allow the user to distribute the labels along the radius of the piechart. So the user could specify if the label should be close to the center or near the border.

**Feature description:** 
Sometimes when you make piecharts with labels, the labels can overlaps if slice are too small.

![image](https://user-images.githubusercontent.com/9820390/204488887-c64648dc-8e50-4d13-aa0f-f96dd71c1114.png)

One way to avoid this problem would be to distribute the labels along the radius, so they would not be overlapping. This could ideally be computed automatically but a first step would be to just allow the users to define where on the radius a label should be. This could be defined in percent, 0% would mean the label is on the center, 100% on the perimeter.

This also could be used when the **ShowValues** or **ShowPercentages** are true.

**Code example:** 
```cs
var values = new int[] {10, 20, 30};
var labels = new string[] {"10", "20", "30"};
var labelsPosition = new float[] {0.33f, 0.66f, 0.99f};

var plt = new ScottPlot.Plot((int)(size.Width * scale), (int)(size.Height * scale));
var pie = plt.AddPie(values);
pie.SliceLabels = labels;
pie.SliceLabelsPositions = labelsRadiusPosition;
```

Is it something that could be possible ?

**Comment by swharden** [comment]
Hi @cpa-level-it, this is a good idea! It has come up in different contexts before (e.g., #1742 and on the Discord #1966). I'll probably solve this by adding an option to show all slices or small slices using _leader lines_, and this open issue will be used to track progress.

However, I'll delay work on this until after #2304 but if someone wants to challenge this before I get to it I would be happy to review a PR!

**Comment by swharden** [comment]
Resolved by #2515
